### ISSUE TITLE
Add underscore validating for release name 

### ISSUE THREAD

**Issue by itielshwartz** [helm/helm#3493]
If trying to deploy a release which contain underscore (_) in its name.

helm give a very unindicative error + create the resources.

cmd: `helm --name not_valid install . --namespace fake --set env=fake`
The following release (named not_valid), got the following error:
```
NAME:   not_valid
Error: getting deployed release "not_valid": release: "not_valid" not found
```
But id did create the resouceres `kubectl get pods --namespace fake`:

```
NAME                                     READY     STATUS    RESTARTS   AGE
fake-redis-deployment-66c7fdf96c-22828   1/1       Running   0          33s
```

After searching for the solution i found this: #1996 

And after checking the logs:
` kubectl logs tiller-deploy-5b9d65c7f-z94tg --namespace kube-system`

I saw this error:
```
[storage/driver] 2018/02/11 22:06:09 create: failed to create: ConfigMap "not_valid.v1" is invalid: metadata.name: Invalid value: "not_valid.v1": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
[tiller] 2018/02/11 22:06:09 warning: Failed to record release not_valid: ConfigMap "not_valid.v1" is invalid: metadata.name: Invalid value: "not_valid.v1": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

I think the correct behaviour is to add this regex validtion before creating all the resources, and kill the deploy before it start.

Would love to take this issue as my first helm commit - is it possible?

Thanks,
Itiel.

**Comment by kvudata** [comment]
/remove-lifecycle rotten
I just hit this issue as well, it's a pretty bad user experience to have no idea why it failed to create the ConfigMap

**Comment by cjbottaro** [comment]
Same. Just ran into this issue. My chart has nothing but very small deployment and service templates.

`helm install --name indexer_app ...` fails.
`helm install --name indexer-app ...` succeeds.

**Comment by cmeyertons** [comment]
This one caused me to waste a couple hours of banging head against wall because other users have so much problems with getting that first install to work.

Just an error message validating release name upfront would be awesome!  That way we don't have to manually clean up resources, etc.

**Comment by cmeyertons** [comment]
Also, why does this have a feature tag?? Absolutely buggy behavior that the tool creates the resources and then fails at the end with a bad error message.

**Comment by manikantabandaru** [comment]
I am also facing same issue , is there any way to resolve the issue? what things i need to check inorder to debug the helm chart ?

**Comment by hickeyma** [comment]
@schollii The validation was added by PR #4883 and made more robust with #5132. It is available since release [2.12.3](https://github.com/helm/helm/releases/tag/v2.12.3).

In Helm 3 will not add the release but returns the error directly from the k8s API.

**Comment by hickeyma** [comment]
Closing as feature has been implemented in PR #4883 and made more robust with #5132.
